src/main/java/ua/okwine/productexpirationdate/service/ImportService.java

```diff
@@ -0,0 +1,128 @@
+import ua.okwine.productexpirationdate.entity.Supplier;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.text.SimpleDateFormat;
+import java.time.LocalDate;
+import java.util.*;
+
+@Service
+@AllArgsConstructor
+@Slf4j
+public class ImportService {
+
+    private final ProductService productService;
+    private final SupplierService supplierService;
+
+    public String saveFile(Model model, MultipartFile file) throws  IOException{
+        String uploadDirectoryPath = "src/main/resources/temp_files";
+        String fullPath = "src/main/resources/temp_files/" + file.getOriginalFilename();
+
+        Path fileNameAndPath = Paths.get(uploadDirectoryPath, file.getOriginalFilename());
+        Files.write(fileNameAndPath, file.getBytes());
+
+        model.addAttribute("msg", "Файл " + file.getOriginalFilename() +
+                " успешно загружен.");
+        return fullPath;
+    }
+
+    public void saveProductFromExcel(Model model, String path) {
+        Map<String, Supplier> providerMap = supplierService.findAllByName();
+        List<Product> productList = excelProductImport(model, path, providerMap);
+
+        productService.saveAll(productList);
+    }
+
+    public  List<Product> excelProductImport(Model model, String excelFilePath, Map<String, Supplier> providerMap) {
+        List <Product> productList = new ArrayList<>();
+        int rowCounter = 1;
+
+        try(FileInputStream inputStream = new FileInputStream(excelFilePath)) {
+            Workbook workbook = new XSSFWorkbook(inputStream);
+            Sheet excelSheet = workbook.getSheetAt(0);
+            Iterator<Row> rowIterator = excelSheet.iterator();
+            SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd");
+            Date tempDate;
+
+            rowIterator.next();
+            while (rowIterator.hasNext()) {
+                rowCounter++;
+                Row nextRow = rowIterator.next();
+                tempDate = null;
+                LocalDate produced = null;
+
+                String vendorCode = nextRow.getCell(1).getStringCellValue();
+                String barCode = nextRow.getCell(2).getStringCellValue();
+                String productName = nextRow.getCell(3).getStringCellValue();
+                tempDate = nextRow.getCell(4).getDateCellValue();
+                if (tempDate != null) {
+                    produced = LocalDate.parse(dateFormat.format(tempDate));
+                }
+
+                tempDate = nextRow.getCell(5).getDateCellValue();
+                LocalDate expirationDate = LocalDate.parse(dateFormat.format(tempDate));
+                Supplier supplier = providerMap.get(nextRow.getCell(6).getStringCellValue());
+
+                productList.add(new Product(vendorCode, barCode, productName,
+                        produced, expirationDate, supplier));
+            }
```

Comment:
It's pretty big method, consider creating some private method with the meaningful name to place the part of the logic there. You can use ctrl+alt+M to move some highlighted part of the code to a private method.